> Now when there's only 1 warning being muted we may explicitly mention in in the commit title.
> Also it would be really useful to have an explanation of why that redefinition happens and maybe even insert an extract from real code which requires that treatment.

The redefinition is caused by this PR : [userspace: memory domain cleanups & arch-specific data support](https://github.com/zephyrproject-rtos/zephyr/pull/27687):

```
1 warning generated.
[106/114] Building C object zephyr/kernel/CMakeFiles/kernel.dir/idle.c.obj
In file included from zephyr/kernel/idle.c:7:
../include\kernel.h:648:26: warning: redefinition of typedef 'k_tid_t' is a C11 feature [-Wtypedef-redefinition]
typedef struct k_thread *k_tid_t;
                         ^
../include\app_memory/mem_domain.h:18:26: note: previous definition is here
typedef struct k_thread *k_tid_t;
```
https://github.com/zephyrproject-rtos/zephyr/blob/04fe7c6eeb7880d2a41d1d9d8e17a3d7d427c9ed/include/kernel.h#L648
https://github.com/zephyrproject-rtos/zephyr/blob/04fe7c6eeb7880d2a41d1d9d8e17a3d7d427c9ed/include/app_memory/mem_domain.h#L18

kernel.h include kernel_includes.h:
https://github.com/zephyrproject-rtos/zephyr/blob/04fe7c6eeb7880d2a41d1d9d8e17a3d7d427c9ed/include/kernel.h#L17
kernel_includes.h include mem_domain.h:
https://github.com/zephyrproject-rtos/zephyr/blob/04fe7c6eeb7880d2a41d1d9d8e17a3d7d427c9ed/include/kernel_includes.h#L45

Maybe we need to do some changes in codes, but it's interesting why only mwdt toolchain will treat this as redefinition warning.

--- a/cmake/compiler/arcmwdt/compiler_flags.cmake
+++ b/cmake/compiler/arcmwdt/compiler_flags.cmake
@@ -31,6 +31,7 @@ set_compiler_property(PROPERTY warning_base
                       -Wno-main-return-type
                       -Wno-unaligned-pointer-conversion
                       -Wno-incompatible-pointer-types-discards-qualifiers
+                      -Wno-typedef-redefinition
 )
 
 check_set_compiler_property(APPEND PROPERTY warning_base -Wno-pointer-sign)

